composables/use-log-event.ts

```diff
@@ -2,17 +2,54 @@ interface EventParams {
   [key: string]: unknown
 }
 
-export default function useLogEvent(eventName: string, eventParams?: EventParams) {
-  const { instance: crisp } = useScriptCrisp()
+export default function useLogEvent(eventName: string, eventParams: EventParams = {}) {
   try {
     useTrackEvent(eventName, eventParams)
   }
   catch {
     console.error(`Failed to track event: ${eventName}`, eventParams)
   }
+
+  const { proxy } = useScriptMetaPixel()
+  try {
+    const eventNameMapping: { [key: string]: string } = {
+      view_item: 'ViewContent',
+      begin_checkout: 'InitiateCheckout',
+      add_to_cart: 'AddToCart',
+      purchase: 'Purchase',
+      search: 'Search',
+    }
+    if (eventNameMapping[eventName]) {
+      const {
+        transaction_id: paymentId,
+        value,
+        currency,
+        items,
+      } = eventParams
+      const eventID = paymentId ? `${eventName}_${paymentId}` : undefined
```

Comment:
Can we keep `Id` consistent?
```suggestion
      const eventId = paymentId ? `${eventName}_${paymentId}` : undefined
```

Reply:
Updated, we still need to do `eventID: eventId` in the actual payload though